src/raft/raft.go

```diff
@@ -308,6 +276,23 @@ func (rf *Raft) HeartBeat(args *HeartBeatRequest, reply *HeartBeatReply) {
 			reply.Good = false
 			reply.Term = rf.term
 
+			// In the case of rf.log[args.PrevLogIndex].Term != args.PrevLogTerm, a good way to speed up is
+			// delete the follower's log from the end with all the entries that having term as rf.log[args.PrevLogIndex].Term
+			// And we should only deal this situation when rf.lastLogIndex() >= args.PrevLogIndex (follower can have much longer logs than args.PrevLogIndex)
+			// Otherwise, if we don't do this, args.PrevLogIndex will only be reduced as 1 with each HeartBeat, that is too slow.
+			if rf.lastLogIndex() >= args.PrevLogIndex {
+				termToDelete := args.PrevLogTerm
+				newEnd := args.PrevLogIndex - 1
+				for ; newEnd > 0; newEnd-- {
+					if rf.log[newEnd].Term <= termToDelete {
+						break
+					}
+				}
+				rf.log = rf.log[:newEnd+1] // remove/cut all logs having rf.log[i].Term > termToDelete
+				// For the case of when rf.log[newEnd].Term < termToDelete, speed up is handled at the leader side
+				// with information of reply.LastLogTerm
+			}
+
```

Comment:
These places for speed-up follower log correction is important for passing `TestFigure8Unreliable2C`